packages/tpetra/core/src/Tpetra_MultiVector_def.hpp

```diff
@@ -3951,6 +3951,13 @@ namespace Tpetra {
 
     // If Case 2 then sum up *this and distribute it to all processes.
     if (Case2) {
+      // FENCE REVIEW - CONFIRMED FAILURE
+      //   Testing: This code is exercised by unit tests.
+      //   GTX960:  Fails with CUDA_LAUNCH_BLOCKING=0 and fence removed. TpetraCore_MultiVector_UnitTests_MPI_4
+      //   White:   Not checked since failure already confirmed on GTX960.
+      //   Plan:    Investigate the reduce - determine failure point - probably we need the fence for a host conversion.
+      //   Notes:
+
       Kokkos::fence();
       this->reduce ();
```

Comment:
This fence can be moved later to just before the call to `allReduceView` [here](https://github.com/Tech-XCorp/Trilinos/blob/0f3cca46a411134d95ee3e5438e87053ceeca675/packages/tpetra/core/src/Tpetra_MultiVector_def.hpp#L4059) and [here](https://github.com/Tech-XCorp/Trilinos/blob/0f3cca46a411134d95ee3e5438e87053ceeca675/packages/tpetra/core/src/Tpetra_MultiVector_def.hpp#L4064).